docs/userguide.rst

```diff
@@ -0,0 +1,204 @@
+infrastructure, so in order to deploy REANA components one needs to have a
+working Kubernetes deployment.
+
+reana-cluster utilizes configuration files of `kubectl -tool <https://kubernetes.io/docs/user-guide/kubectl-overview/>`_
+to set up a connection to Kubernetes infrastructure, so in order
+to use reana-cluster one needs to have kubectl installed
+and configured properly.
+
+
+Test reana-cluster using Minikube
+---------------------------------
+
+Easiest way to test reana-cluster without dedicating time to
+setting up a real Kubernetes infrastructure is to use
+`Minikube <https://kubernetes.io/docs/getting-started-guides/minikube/>`_,
+which is a spins up a working Kubernetes deployment in a virtual machine.
+
+Once you have installed kubectl and Minikube, start Minikube by running:
+
+.. code-block:: console
+
+   $ minikube start --kubernetes-version="v1.6.4"
+   ...
+   minikube: Running
+   cluster: Running
+   kubectl: Correctly Configured: pointing to minikube-vm at 192.168.99.100
+
+.. note::
+   Currently REANA supports only Kubernetes v1.6.4
+
+As seen from the output, Minikube startup routines should
+configure kubectl to interact with the newly created
+virtual machine, but it best to test that kubectl is indeed
+configured properly:
+
+.. code-block:: console
+
+   $ kubectl get all
+   NAME             CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
+   svc/kubernetes   10.0.0.1     <none>        443/TCP   3d
+
+
+Install reana-cluster cli tool
+------------------------------
+
+reana-cluster command line interface tool is not yet released
+in PyPI, so install it from our GitHub repotory:
+
+.. code-block:: console
+
+   $ pip install \
+     -e 'git+https://github.com/reanahub/reana-cluster.git@master#egg=reana-cluster'
+
+
+.. _configure:
+
+Configure REANA cluster
+-----------------------
+
+Main function of reana-cluster command line tool is to
+initialize a working REANA cluster, ready to run workflows
+you submit to it using reana-client.
+
+In order to achieve this reana-cluster needs to know how
+REANA cluster should be set up; e.g. what versions of REANA
+components should be deployed and how the configuration of each
+component should be set up.
+
+reana-client expects to get information via REANA cluster
+specification file.
+
+The specifications file is written in YAML syntax, but since
+reana-cluster is still work-in-progress the structure of
+REANA specifications file might change rapidly.
+We therefore suggest that you download our default
+configuration file from the root of our GitHub repository:
+
+.. code-block:: console
+
+   $ wget https://raw.githubusercontent.com/reanahub/reana-cluster/master/reana-cluster.yaml
+
+Default REANA cluster specifications file deploys latest
+released versions of all REANA components in their
+default configuration. Please note that default specifications file
+is intended for evaluation, not for production deployments.
+
+Example:
```

Comment:
Example needs to be updated before releasing since it has changed.

Reply:
Yes, we should update the docs after testing. Note that the main purpose of this PR is the restructuring.  So I just introduced new sections but kept the original text unchanged... The text changes should come afterwards.